updater/aws_test.go

```diff
@@ -15,6 +15,260 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestFilterAvailableUpdates(t *testing.T) {
+	instances := []instance{
+		{
+			instanceID:          "inst-id-1",
+			containerInstanceID: "cont-inst-1",
+		},
+		{
+			instanceID:          "inst-id-2",
+			containerInstanceID: "cont-inst-2",
+		},
+		{
+			instanceID:          "inst-id-3",
+			containerInstanceID: "cont-inst-3",
+		},
+		{
+			instanceID:          "inst-id-4",
+			containerInstanceID: "cont-inst-4",
+		},
+		{
+			instanceID:          "inst-id-5",
+			containerInstanceID: "cont-inst-5",
+		},
+	}
+	expected := []instance{
+		{
+			instanceID:          "inst-id-1",
+			containerInstanceID: "cont-inst-1",
+			bottlerocketVersion: "v1.0.5",
+		},
+		{
+			instanceID:          "inst-id-2",
+			containerInstanceID: "cont-inst-2",
+			bottlerocketVersion: "v1.0.5",
+		},
+		{
+			instanceID:          "inst-id-5",
+			containerInstanceID: "cont-inst-5",
+			bottlerocketVersion: "v1.0.5",
+		},
+	}
+	responses := map[string]string{
+		"inst-id-1": "{\"update_state\": \"Available\", \"active_partition\": { \"image\": { \"version\": \"v1.0.5\"}}}",
```

Comment:
(multiple occurrences)
nit: it can be useful to use \` to have strings without escapes when writing literal JSON
```suggestion
		"inst-id-1": `{"update_state": "Available", "active_partition": { "image": { "version": "v1.0.5"}}}`,
```